calculator.rb

```diff
@@ -0,0 +1,97 @@
+puts "Thank you for choosing my calculator"
+answer = 0
+
+def is_numeric(s)
+  # This begin / rescue block is for exception catching.
+  # We need it because we're trying to convert string to float and it might fail if string contains non-numeric characters
+  # on successful conversion - it will return true, else false
+  begin
+    Float(s)
+  rescue
+    false # not numeric
+  else
+    true # numeric
+  end
+end
+
+
+def get_numeric_input
+  while user_input = gets.chomp # loop while getting user input
+    isInputOK = is_numeric(user_input)
+    case isInputOK
+    when true
+      break # make sure to break so you don't ask again
+    else
+      puts "Please enter valid number"
+    end
+  end
+  return user_input
+end
+
+# calculate addition
+def add
+  puts "Please enter your first number"
+  num1 = get_numeric_input.to_f
+  puts "Please enter your second number"
+  num2 = get_numeric_input.to_f
```

Comment:
These four lines are repeated in each of these methods. Could you consolidate this somehow?